pkg/distributor/push.go

```diff
@@ -190,9 +226,18 @@ func toHTTPStatus(ctx context.Context, pushErr error, limits *validation.Overrid
 	return http.StatusInternalServerError
 }
 
-func addHeaders(w http.ResponseWriter, err error) {
+func addHeaders(w http.ResponseWriter, err error, r *http.Request, code int, retryCfg *RetryConfig, logger glog.Logger) {
 	var doNotLogError middleware.DoNotLogError
 	if errors.As(err, &doNotLogError) {
 		w.Header().Set(server.DoNotLogErrorHeaderKey, "true")
 	}
+
+	if (code == http.StatusTooManyRequests || code/100 == 5) && retryCfg != nil {
+		var retrySeconds string
+		if retryCfg.Enabled {
+			retrySeconds = calculateRetryAfter(r, retryCfg)
+			w.Header().Set("Retry-After", retrySeconds)
+		}
+		logger.Log("msg", "set retry-after", "enabled", retryCfg.Enabled, "retry-after", retrySeconds, "maxAllowedAttemps", retryCfg.MaxAllowedAttempts)
```

Comment:
would be nice to see the value of retry-attempt here too

Reply:
can we tag the trace with retry-after and retry-attempt? This way you can find traces which have a retry-attempt set for example or more than X, or not a number, etc

Reply:
Can we combine this log line with the one which logs "push error"? I think two log lines per response is confusing and it will be easier to make sense of the two if they are grouped on the same line

Reply:
also the value of `retry-after` will be hard to use in loki. Should we use something like `retry_after` or `retryAfter` to match the other field (or rename the other to `max_allowed_attempts`)?

Reply:
i want to keep things easy retry_after is fine and seems reasonable. Add to trace is a good idea, i can just use span logger, but combine the 2 log lines into 1, we need to pass the variables around and it would be included in all logs even the retry after is not enabled. I am not sure whether it is more noisy or not.

Reply:
will add them to traces instead